flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -586,7 +586,7 @@ private StreamGraph generateInternal(List<StreamTransformation<?>> transformatio
 				transform.getOutputType(),
 				transform.getName());
 
-		if (transform.getStateKeySelector1() != null) {
+		if (transform.getStateKeySelector1() != null || transform.getStateKeySelector2() != null) {
```

Comment:
I think we need some test that using keyed state actually works when only one of the inputs is keyed. I think we need an `ITCase` for that somewhere.

Reply:
Does this, in combination with the comment below on the `DataStreamTest` mean that you want to move the `DataStreamTest` test in a separate `ITCase`?